x/evm/keeper/genesis.go

```diff
@@ -0,0 +1,75 @@
+package keeper
+
+import (
+	"github.com/axelarnetwork/axelar-core/utils"
+	"github.com/axelarnetwork/axelar-core/x/evm/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (k baseKeeper) InitGenesis(ctx sdk.Context, state types.GenesisState) {
+	for _, chain := range state.Chains {
+		ck := k.ForChain(chain.Params.Chain).(chainKeeper)
```

Comment:
interesting. I like it how you are exposing private method here.

Reply:
I thought a while about how to best do this. We cannot change the return type of `ForChain` because it is also used in other places, but since this is the `baseKeeper` itself in control here I think it is ok to hard cast the `chainKeeper`